lib/src/metta/runner/builtin_mods/fileio.rs

```diff
@@ -0,0 +1,156 @@
+#[derive(Debug)]
+pub(crate) struct FileioModLoader;
+
+impl ModuleLoader for FileioModLoader {
+    fn load(&self, context: &mut RunContext) -> Result<(), String> {
+
+        let space = DynSpace::new(GroundingSpace::new());
+        context.init_self_module(space, None);
+
+        let _ = self.load_tokens(context.module(), context.metta.clone())?;
+
+        let parser = SExprParser::new(FILEIO_METTA);
+        context.push_parser(Box::new(parser));
+
+        Ok(())
+    }
+
+    fn load_tokens(&self, target: &MettaMod, _metta: Metta) -> Result<(), String> {
+        let mut tref = target.tokenizer().borrow_mut();
+
+        tref.register_function(GroundedFunctionAtom::new(
+            r"file-open".into(),
+            Atom::expr([ARROW_SYMBOL, ATOM_TYPE_STRING, ATOM_TYPE_STRING, ATOM_TYPE_FILE_HANDLE]),
+            file_open));
+
+        tref.register_function(GroundedFunctionAtom::new(
+            r"file-read".into(),
+            Atom::expr([ARROW_SYMBOL, ATOM_TYPE_FILE_HANDLE, ATOM_TYPE_STRING]),
+            file_read));
+
+        tref.register_function(GroundedFunctionAtom::new(
+            r"file-write".into(),
+            Atom::expr([ARROW_SYMBOL, ATOM_TYPE_FILE_HANDLE, ATOM_TYPE_STRING, UNIT_ATOM]),
+            file_write));
+
+        Ok(())
+    }
+}
+
+fn file_open(args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+    let arg_error = || ExecError::from("file-open expects filepath and options in form of string atoms as an argument");
+    let filepath = args.get(0).and_then(Str::from_atom).ok_or_else(arg_error)?;
+    let options = args.get(1).and_then(Str::from_atom).ok_or_else(arg_error)?;
+
+    let fhandler = FileHandle::open(filepath.as_str(), options.as_str())?;
+    Ok(vec![Atom::gnd(fhandler)])
+}
+
+fn file_read(args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+    let arg_error = || ExecError::from("file-read expects filehandle as an argument");
+    let filehandle = args.get(0).ok_or_else(arg_error)?.into();
+    let filehandle = Atom::as_gnd::<FileHandle>(filehandle).ok_or("file-read expects filehandle as an argument")?;
+
+    let message = filehandle.read_as_str();
+    Ok(vec![Atom::gnd(Str::from_string(message))])
+}
+
+fn file_write(args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+    let arg_error = || ExecError::from("file-write expects filehandle and content (string atom) as an arguments");
+    let filehandle = args.get(0).ok_or_else(arg_error)?.into();
+    let filehandle = Atom::as_gnd::<FileHandle>(filehandle).ok_or("file-write expects filehandle and content (string atom) as an arguments")?;
+    let content = args.get(1).and_then(Str::from_atom).ok_or_else(arg_error)?;
+
+    filehandle.write(content.as_str());
+    unit_result()
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::metta::*;
+    use crate::metta::runner::run_program;
+
+    #[test]
+    fn test_filehandle() {
+        let program = "
+            !(import! &self fileio)
+            !(bind! &fhandle (file-open \"test.txt\" \"rwc\"))
```

Comment:
I would recommend using temporary file name instead. You can get it using this library https://docs.rs/tempfile/latest/tempfile/struct.NamedTempFile.html